homeassistant/components/syncthing/sensor.py

```diff
@@ -0,0 +1,224 @@
+"""Support for monitoring the Syncthing instance."""
+
+import logging
+
+import aiosyncthing
+
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (
+    DOMAIN,
+    FOLDER_PAUSED_RECEIVED,
+    FOLDER_SENSOR_ALERT_ICON,
+    FOLDER_SENSOR_DEFAULT_ICON,
+    FOLDER_SENSOR_ICONS,
+    FOLDER_SUMMARY_RECEIVED,
+    SCAN_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+    STATE_CHANGED_RECEIVED,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Syncthing sensors."""
+
+    name = config_entry.data[CONF_NAME]
+    syncthing = hass.data[DOMAIN][name]
+
+    try:
+        config = await syncthing.system.config()
+        version = await syncthing.system.version()
+        dev = [
+            FolderSensor(
+                hass,
+                syncthing,
+                name,
+                folder["id"],
+                folder["label"],
+                version["version"],
+            )
+            for folder in config["folders"]
+        ]
+
+        async_add_entities(dev)
+    except aiosyncthing.exceptions.SyncthingError as exception:
+        raise PlatformNotReady from exception
+
+
+class FolderSensor(Entity):
+    """A Syncthing folder sensor."""
+
+    def __init__(self, hass, syncthing, name, folder_id, folder_label, version):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.